### PR TITLE
Clear up the Licensing requirements

### PR DESCRIPTION
**Pull request by Luspin**
I think the requirements are a bit misleading, so I changed the text a bit to match:
https://docs.microsoft.com/en-us/azure/active-directory/users-groups-roles/groups-naming-policy

### CODE REVIEW

**Comment by mypil** [comment]
@nenonix @JohanFreelancer9 - kindly do the copy editor review for this PR. Thanks.

File: Office365-Admin/create-groups/groups-naming-policy.md [hunk 1]
```diff
@@ -40,10 +40,8 @@ The group naming policy consists of the following features:
     
 ## Licensing requirements
 
-To use the Groups naming policy feature, the following people need an Azure Active Directory Premium P1 license or Azure AD Basic EDU license:
-- Everyone who is a member of the group.
-- The person who creates the group.
-- The admin who creates the Groups naming policy  
+Using Azure AD naming policy for Office 365 groups requires that you possess but not necessarily assign an Azure Active Directory Premium P1 license or Azure AD Basic EDU license for each unique user (including guests) that is a member of one or more Office 365 groups.
```
Comment (nenonix) [inline Office365-Admin/create-groups/groups-naming-policy.md]:
```suggestion
Using Azure AD naming policy for Office 365 groups requires that you possess, but not necessarily assign, an Azure Active Directory Premium P1 license or Azure AD Basic EDU license for each unique user that is a member of one or more Office 365 groups. This includes guest users and is also required for the administrator who creates the Groups naming policy.
```

File: Office365-Admin/create-groups/groups-naming-policy.md [hunk 2]
```diff
@@ -40,10 +40,8 @@ The group naming policy consists of the following features:
...
+This is also required for the administrator that creates the Groups naming policy.
```
Comment (nenonix) [inline Office365-Admin/create-groups/groups-naming-policy.md]:
```suggestion
```

File: Office365-Admin/create-groups/groups-naming-policy.md (hunk 1 above)
Comment (Luspin) [reply Office365-Admin/create-groups/groups-naming-policy.md]: Looks good to me, let's go for it.

**Comment by cmcatee-MSFT** [comment]
@MikePlumleyMSFT Can you please review the changes?

**Comment by MikePlumleyMSFT** [comment]
I'm working on clarifying the licensing requirements for several of the Groups features. Waiting to hear back on that before we address this PR.

**Comment by mypil** [comment]
@MikePlumleyMSFT - Do you have any updates on this? Thank you.

**Comment by mypil** [comment]
@MikePlumleyMSFT - We would like to follow up. Do you have any updates?

**Comment by cmcatee-MSFT** [comment]
Per @MikePlumleyMSFT , still waiting on a response from the product team on this.

**Pull request merged**